app/src/main/java/com/example/woloxapp/ui/login/LoginFragment.kt

```diff
@@ -18,4 +28,52 @@ class LoginFragment : Fragment() {
         _binding = LoginFragmentBinding.inflate(inflater, container, false)
         return binding.root
     }
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        loginViewModel = ViewModelProvider(this)[LoginViewModel::class.java]
+        loginViewModel.getUserModel()
+        getObservers()
+        validCredentials()
+        with(binding) {
+            loginBtn.setOnClickListener {
+                val emailText = userName.text.toString()
+                val passwordText = password.text.toString()
+                loginViewModel.fieldsValidation(emailText, passwordText)
+            }
+            signupBtn.setOnClickListener {
+                // TODO: navigate to signupScreen
+            }
+        }
+    }
+
+    private fun getObservers() {
+        val toast = Toast.makeText(context, getString(R.string.required_fields), Toast.LENGTH_SHORT)
+        toast.setGravity(Gravity.TOP, toastHorizontal, toastVertical)
+        loginViewModel.emptyFieldsError.observe(viewLifecycleOwner) {
+            it?.let { toast.show() }
+        }
+        loginViewModel.validEmail.observe(viewLifecycleOwner) {
+            if (it == false) {
+                binding.userName.error = getString(R.string.invalid_username)
+            }
+        }
+        loginViewModel.userEmail.observe(viewLifecycleOwner) {
+            it.let {
+                binding.userName.setText(it)
+            }
+        }
+        loginViewModel.userPassword.observe(viewLifecycleOwner) {
+            it.let {
+                binding.password.setText(it)
+            }
+        }
+    }
+
+    private fun validCredentials() {
+        loginViewModel.validEmail.observe(viewLifecycleOwner) {
+            if (it == true) {
```

Comment:
this is okay, but keep in mind that when you add real credentials validation this should change